vms/platformvm/state/camino_state.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2022, Chain4Travel AG. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import (
+	"math"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/vms/platformvm/lock"
+)
+
+func (s *state) LockedUTXOs(txID ids.ID, addresses []ids.ShortID, lockState lock.State) ([]*avax.UTXO, error) {
+	txMatch := func(lockID lock.IDs) bool {
+		switch lockState {
+		case lock.LockStateDeposited:
+			return lockID.DepositTxID == txID
```

Comment:
Since we now have introduced locking of one TX == both states, we must also check for both if state is depositedBonded